benches/generic.rs

```diff
@@ -0,0 +1,553 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use revision::prelude::*;
+use std::borrow::Cow;
+use std::collections::{BTreeMap, BTreeSet, BinaryHeap, HashMap, HashSet};
+use std::time::Duration;
+
+// A comprehensive struct that uses many of the types supported by revision
+#[derive(Debug)]
+#[revisioned(revision = 1)]
+struct ComplexData {
+	// Primitive integers
+	tiny_int: u8,
+	small_int: u16,
+	medium_int: u32,
+	large_int: u64,
+	huge_int: u128,
+	size_int: usize,
+
+	// Signed integers
+	tiny_signed: i8,
+	small_signed: i16,
+	medium_signed: i32,
+	large_signed: i64,
+	huge_signed: i128,
+	size_signed: isize,
+
+	// Floating point
+	float32: f32,
+	float64: f64,
+
+	// Character and boolean
+	character: char,
+	flag: bool,
+
+	// Strings and collections
+	text: String,
+	bytes: Vec<u8>,
+	numbers: Vec<i32>,
+
+	// Additional vector types for comprehensive testing
+	boolean_flags: Vec<bool>,
+	tiny_signed_vec: Vec<i8>,
+	small_unsigned_vec: Vec<u16>,
+	large_numbers_vec: Vec<i64>,
+	huge_numbers_vec: Vec<u128>,
+	float_values_vec: Vec<f32>,
+	double_precision_vec: Vec<f64>,
+	string_list: Vec<String>,
+
+	// Arrays
+	small_array: [u8; 4],
+	medium_array: [i32; 16],
+
+	// Optional values
+	maybe_text: Option<String>,
+	maybe_number: Option<u64>,
+
+	// Boxed values
+	boxed_data: String,
+
+	// Tuples
+	pair: (String, i32),
+	triple: (u8, u16, u32),
+	quad: (bool, char, f32, String),
+	quint: (i8, i16, i32, i64, String),
+
+	// Collections with complex keys/values
+	string_to_int: HashMap<String, i32>,
+	ordered_map: BTreeMap<String, Vec<u8>>,
+	unique_strings: HashSet<String>,
+	ordered_numbers: BTreeSet<i32>,
+	priority_queue: BinaryHeap<i32>,
+
+	// Results
+	success_result: Result<String, i32>,
+	error_result: Result<u8, String>,
+
+	// Duration
+	time_duration: Duration,
+
+	// Cow (clone-on-write)
+	#[allow(clippy::owned_cow)]
+	borrowed_or_owned: Cow<'static, String>,
```

Comment:
Shouldn't this be `Cow<'static, str>`?

Reply:
`revision` only supports `Cow<'static, T> where T: sized` at the moment. I'll do that in a separate PR.